homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,261 @@
+            # We need to create a new instance to reconnect.
+            try:
+                LOGGER.debug("Create SamsungRemote")
+                self._remote = Remote(self.config.copy())
+            # This is only happening when the auth was switched to DENY
+            # A removed auth will lead to socket timeout because waiting for auth popup is just an open socket
+            except AccessDenied:
+                self._notify_callback()
+                raise
+        return self._remote
+
+    def _send_key(self, key):
+        """Send the key using legacy protocol."""
+        self._get_remote().control(key)
+
+
+def _hide_token(config):
+    if config[CONF_TOKEN]:
+        copy = config.copy()
+        copy[CONF_TOKEN] = "*****"
+        return copy
+    return config
+
+
+class SamsungTVWSBridge(SamsungTVBridge):
+    """The Bridge for WebSocket TVs."""
+
+    def __init__(self, method, host, port, token=None):
+        """Initialize Bridge."""
+        super().__init__(method, host, port, token)
+
+    def try_connect(self):
+        """Try to connect to the Websocket TV."""
+        for self.port in (8001, 8002):
+            config = {
+                CONF_NAME: VALUE_CONF_NAME,
+                CONF_HOST: self.host,
+                CONF_METHOD: self.method,
+                CONF_PORT: self.port,
+                # We need this high timeout because waiting for auth popup is just an open socket
+                CONF_TIMEOUT: 31,
+                CONF_TOKEN: self.token,
+            }
+            try:
+                LOGGER.debug("Try config: %s", _hide_token(config))
+                with SamsungTVWS(
+                    host=self.host,
+                    port=self.port,
+                    token=self.token,
+                    timeout=config[CONF_TIMEOUT],
+                    name=config[CONF_NAME],
+                ) as remote:
+                    remote.open()
+                    self.token = remote.token
+                LOGGER.debug("Working config: %s", _hide_token(config))
+                return RESULT_SUCCESS
+            except WebSocketException:
+                LOGGER.debug("Working but unsupported config: %s", _hide_token(config))
+                return RESULT_NOT_SUPPORTED
+            except (OSError, ConnectionFailure) as err:
+                LOGGER.debug("Failing config: %s, error: %s", _hide_token(config), err)
+
+        return RESULT_NOT_SUCCESSFUL
+
+    def _send_key(self, key):
+        """Send the key using websocket protocol."""
+        if key == "KEY_POWEROFF":
+            key = "KEY_POWER"
+        self._get_remote().send_key(key)
+
+    def _get_remote(self):
+        """Create or return a remote control instance."""
+        if self._remote is None:
+            # We need to create a new instance to reconnect.
+            try:
+                LOGGER.debug("Create SamsungTVWS")
+                self._remote = SamsungTVWS(
+                    host=self.config[CONF_HOST],
```

Comment:
Here we should use the corresponding instance attributes directly, not `self.config`.

Reply:
done